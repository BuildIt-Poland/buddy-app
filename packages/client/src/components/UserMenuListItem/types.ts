import Maybe from 'graphql/tsutils/Maybe';
import { Buddy, Newbie } from '@buddy-app/schema';

export type UserMenuListItemProps = {
  user: Maybe<Newbie> | Maybe<Buddy>;
  onItemClick: (id: string) => void;
};
