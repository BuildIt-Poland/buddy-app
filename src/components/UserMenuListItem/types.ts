import Maybe from 'graphql/tsutils/Maybe';
import { Buddy, Newbie, User } from 'buddy-app-schema';

export type UserMenuListItemProps = {
  user: Maybe<Newbie> | Maybe<Buddy>;
  onItemClick: (id: string) => void;
};
