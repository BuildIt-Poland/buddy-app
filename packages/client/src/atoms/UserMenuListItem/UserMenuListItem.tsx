import React from 'react';
import ListItem from '@material-ui/core/ListItem';
import Box from '@material-ui/core/Box';
import Typography from '@material-ui/core/Typography';
import Avatar from 'atoms/Avatar';
import { User } from '@buddy-app/schema';
import { makeStyles } from '@material-ui/core/styles';
import { UserMenuListItemProps } from './types';

const useStyles = makeStyles(theme => ({
  avatar: {
    height: '5.5rem',
    marginRight: theme.spacing(1.5),
  },
  list: {
    display: 'flex',
    alignItems: 'center',
    width: '100%',
  },
  listItem: {
    paddingTop: 0,
    paddingBottom: 0,
    paddingLeft: theme.spacing(2),
    paddingRight: theme.spacing(2),
  },
}));

const UserMenuListItem: React.FC<UserMenuListItemProps> = props => {
  const { user, onItemClick } = props;
  const { avatar, list, listItem } = useStyles();
  /* istanbul ignore next */
  const handleClick = () => onItemClick((user as User).id);

  return (
    <ListItem button onClick={handleClick} className={listItem}>
      <Box className={list}>
        <Box className={avatar}>
          <Avatar type={'small'} imgSrc={(user as User).photo} />
        </Box>
        <Typography component='p' variant='body2'>
          {(user as User).name}
        </Typography>
      </Box>
    </ListItem>
  );
};

export default UserMenuListItem;
