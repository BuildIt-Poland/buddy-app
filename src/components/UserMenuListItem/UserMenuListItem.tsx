import React from 'react';
import ListItem from '@material-ui/core/ListItem';
import Box from '@material-ui/core/Box';
import Typography from '@material-ui/core/Typography';
import Avatar from 'components/Avatar';
import theme from 'styles/theme';
import { User } from 'buddy-app-schema';
import { makeStyles } from '@material-ui/core/styles';
import { UserMenuListItemProps } from './types';

const useStyles = makeStyles({
  avatar: {
    width: '9rem',
    height: '5.5rem',
  },
  list: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    width: '100%',
    justifyContent: 'space-between',
  },
  listItem: {
    paddingTop: 0,
    paddingBottom: 0,
    paddingLeft: theme.spacing(2),
    paddingRight: theme.spacing(2),
  },
});

const UserMenuListItem: React.FC<UserMenuListItemProps> = props => {
  const { user, onItemClick } = props;
  const { avatar, list, listItem } = useStyles();

  return (
    <ListItem
      button
      onClick={() => onItemClick((user as User).id)}
      className={listItem}>
      <Box className={list}>
        <Typography component='p' variant='body2'>
          {(user as User).name}
        </Typography>
        <Box className={avatar}>
          <Avatar type={'small'} imgSrc={(user as User).photo} />
        </Box>
      </Box>
    </ListItem>
  );
};

export default UserMenuListItem;
