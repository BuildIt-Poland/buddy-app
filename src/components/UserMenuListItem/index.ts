import UserMenuListItem from './UserMenuListItem';

export default UserMenuListItem;
