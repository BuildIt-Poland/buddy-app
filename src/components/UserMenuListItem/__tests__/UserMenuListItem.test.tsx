import React from 'react';
import { create } from 'react-test-renderer';
import UserMenuListItem from '../UserMenuListItem';

jest.mock('@material-ui/core/ListItem', () => 'ListItem');
jest.mock('@material-ui/core/Box', () => 'Box');
jest.mock('@material-ui/core/Typography', () => 'Typography');
jest.mock('../../Avatar', () => 'Avatar');

describe('Component - UserMenuListItem', () => {
  test('renders correctly', () => {
    const component = create(
      <UserMenuListItem
        user={{ id: 'id', name: 'Tom', photo: 'url' }}
        onItemClick={() => null}
      />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
