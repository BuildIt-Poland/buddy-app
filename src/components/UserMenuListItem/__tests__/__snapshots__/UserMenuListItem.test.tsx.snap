// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - UserMenuListItem renders correctly 1`] = `
<ListItem
  button={true}
  className="makeStyles-listItem-3"
  onClick={[Function]}
>
  <Box
    className="makeStyles-list-2"
  >
    <Typography
      component="p"
      variant="body2"
    >
      Tom
    </Typography>
    <Box
      className="makeStyles-avatar-1"
    >
      <Avatar
        imgSrc="url"
        type="small"
      />
    </Box>
  </Box>
</ListItem>
`;
